kevm-pyk/src/tests/integration/test_conformance.py

```diff
@@ -57,66 +43,142 @@ def _assert_exit_code_zero(pattern: Pattern) -> None:
+assert len(ALL_VM_TESTS) == len(VM_TESTS) + len(REST_VM_TESTS)
 
 
 @pytest.mark.parametrize(
-    'test_file',
+    'test_file, test_id',
     VM_TESTS,
-    ids=[str(test_file.relative_to(VM_TEST_DIR)) for test_file in VM_TESTS],
+    ids=[f'{test_file}:{test_id}' for test_file, test_id in VM_TESTS],
 )
-def test_vm(test_file: Path) -> None:
-    _test(test_file, 'DEFAULT', 'VMTESTS', 1, True)
+def test_vm(get_gst_data: Callable[[Path, str], dict[str, Any]], test_file: Path, test_id: str) -> None:
+    # Given
+    gst_file = TEST_DIR / test_file
+    gst_input = get_gst_data(gst_file, test_id)
+
+    # When
+    res = interpret(gst_input, 'DEFAULT', 'VMTESTS', 1, True, check=False)
+
+    # Then
+    _assert_exit_code_zero(res)
 
 
 @pytest.mark.skip(reason='failing / slow VM tests')
 @pytest.mark.parametrize(
-    'test_file',
+    'test_file, test_id',
     REST_VM_TESTS,
-    ids=[str(test_file.relative_to(VM_TEST_DIR)) for test_file in REST_VM_TESTS],
+    ids=[f'{test_file}:{test_id}' for test_file, test_id in REST_VM_TESTS],
 )
-def test_rest_vm(test_file: Path) -> None:
-    _test(test_file, 'DEFAULT', 'VMTESTS', 1, True)
+def test_rest_vm(get_gst_data: Callable[[Path, str], dict[str, Any]], test_file: Path, test_id: str) -> None:
+    # Given
+    gst_file = TEST_DIR / test_file
+    gst_input = get_gst_data(gst_file, test_id)
+
+    # When
+    res = interpret(gst_input, 'DEFAULT', 'VMTESTS', 1, True, check=False)
 
+    # Then
+    _assert_exit_code_zero(res)
+
+
+EXCLUDED_TESTS = SKIPPED_TESTS.union(ALL_VM_TESTS)
 
 BCHAIN_TEST_DIR: Final = TEST_DIR / 'BlockchainTests/GeneralStateTests'
-ALL_BCHAIN_TESTS: Final = tuple(BCHAIN_TEST_DIR.glob('**/*.json'))
-BCHAIN_TESTS: Final = tuple(
-    test_file for test_file in ALL_BCHAIN_TESTS if test_file not in (list(SKIPPED_TESTS) + list(ALL_VM_TESTS))
-)
-REST_BCHAIN_TESTS: Final = tuple(
-    test_file for test_file in ALL_BCHAIN_TESTS if test_file in SKIPPED_TESTS and test_file not in ALL_VM_TESTS
-)
+ALL_BCHAIN_TESTS_CSV: Final = REPO_ROOT / 'tests/all_bchain_tests.csv'
+BCHAIN_TEST_FILES: Final = tuple(BCHAIN_TEST_DIR.glob('**/*.json'))
+ALL_BCHAIN_TESTS: Final = read_csv_file(ALL_BCHAIN_TESTS_CSV)
+BCHAIN_TESTS: Final = tuple(test for test in ALL_BCHAIN_TESTS if test not in EXCLUDED_TESTS)
+REST_BCHAIN_TESTS: Final = tuple(test for test in ALL_BCHAIN_TESTS if test in EXCLUDED_TESTS)
+assert len(ALL_BCHAIN_TESTS) == len(BCHAIN_TESTS) + len(REST_BCHAIN_TESTS)
+
+
+@pytest.fixture(scope='session')
+def gst_data() -> dict[Path, dict[str, Any]]:
+    return {file: json.loads(file.read_text()) for file in BCHAIN_TEST_FILES + VM_TEST_FILES}
+
+
+@pytest.fixture(scope='session')
+def get_gst_data(gst_data: dict[Path, dict[str, Any]]) -> Callable[[Path, str], dict[str, Any]]:
+    def _get_gst_data(test_file: Path, test_id: str) -> dict[str, Any]:
+        return {test_id: gst_data[test_file][test_id]}
+
+    return _get_gst_data
```

Comment:
Let's extract a class, e.g. `GSTCache`, that does the loading and caching lazily (i.e. it reads each file once on first access). This will:
- Hopefully reduce memory consumption significantly for each process, without significant overhead in execution time.
- Remove the overhead in start-up time.